### ISSUE TITLE
Serializing NbtTag results in a SerializationException

### ISSUE THREAD

**Issue by Gaming32** [BenWoodworth/knbt#14]
If you attempt to serialize an `NbtTag` and have a subclass of that tag, a `SerializationException` occurs.
```
Exception in thread "main" kotlinx.serialization.SerializationException: Class 'NbtCompound' is not registered for polymorphic serialization in the scope of 'NbtTag'.
Mark the base class as 'sealed' or register the serializer explicitly.
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:102)
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:113)
	at kotlinx.serialization.PolymorphicSerializerKt.findPolymorphicSerializer(PolymorphicSerializer.kt:109)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:229)
	at kotlinx.serialization.encoding.Encoder$DefaultImpls.encodeNullableSerializableValue(Encoding.kt:299)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableValue(AbstractEncoder.kt:18)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableElement(AbstractEncoder.kt:90)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeNullableSerializableElement(StreamingJsonEncoder.kt:154)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.internal.NullableSerializer.serialize(NullableSerializer.kt:23)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at kotlinx.serialization.internal.CollectionLikeSerializer.serialize(CollectionSerializers.kt:69)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at io.github.gaming32.mckt.PlayerData.write$Self(world.kt:494)
	at io.github.gaming32.mckt.PlayerData$$serializer.serialize(world.kt:494)
	at io.github.gaming32.mckt.PlayerData$$serializer.serialize(world.kt:494)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.json.internal.JsonStreamsKt.encodeByWriter(JsonStreams.kt:28)
	at kotlinx.serialization.json.JvmStreamsKt.encodeToStream(JvmStreams.kt:25)
	... 19 definitely unrelated frames
```

**Comment by Gaming32** [comment]
Maybe this is correct that it should error here, but this exception is not a very useful error message for users of knbt.

**Comment by BenWoodworth** [comment]
Good callout!

I know the version of kotlinx.serialization that knbt v0.11 uses didn't properly support marking sealed interfaces `@Serializable` (which works seamlessly with sealed classes`) and I had to special case `NbtTag` serialization. Might just be a matter of updating kotlinx.serialization to v1.4.0 so it has [the fix](https://github.com/Kotlin/kotlinx.serialization/pull/1968), or maybe compiling against a newer version of Kotlin with a fixed serializer compiler plugin

Just added some tests locally to see if I could reproduce this, and I did get a failure with something I expected to work, though the error is different. That's something I'll have to fix

![image](https://user-images.githubusercontent.com/1829157/189542258-1cded112-66ab-4fb4-95e5-8d9fea1f377a.png)
